pkg/scheduler/extender_test.go

```diff
@@ -388,3 +390,85 @@ func TestIsInterested(t *testing.T) {
 		})
 	}
 }
+
+// TestRestoreNumPDBViolations test functions convertToNodeNameToMetaVictims and HTTPExtender.convertToNodeNameToVictims in HTTPExtender.ProcessPreemption
+// Ensure the NumPDBViolations of each node is returned from external server.
+func TestRestoreNumPDBViolations(t *testing.T) {
+	se := &HTTPExtender{}
+	nodes := []string{"node1", "node2"}
+	podsInNodeList := []*v1.Pod{
+		{ObjectMeta: metav1.ObjectMeta{Name: "pod1", UID: "uid1"}},
+		{ObjectMeta: metav1.ObjectMeta{Name: "pod2", UID: "uid2"}},
+		{ObjectMeta: metav1.ObjectMeta{Name: "pod3", UID: "uid3"}},
+		{ObjectMeta: metav1.ObjectMeta{Name: "pod4", UID: "uid4"}},
+	}
```

Comment:
These steps along with other nodes instantiations should be abstracted into the body of each sub-test.